targets/tofino/base/expression_resolver.cpp

```diff
@@ -23,7 +24,630 @@ const IR::Expression *TofinoBaseExpressionResolver::processTable(const IR::P4Tab
 
 // Provides implementations of Tofino externs.
 namespace TofinoBaseExterns {
-const ExternMethodImpls EXTERN_METHOD_IMPLS({});
+
+const ExternMethodImpls EXTERN_METHOD_IMPLS(
+    {// -----------------------------------------------------------------------------
+     // CHECKSUM
+     // -----------------------------------------------------------------------------
+     // Tofino checksum engine can verify the checksums for header-only checksums
+     // and calculate the residual (checksum minus the header field
+     // contribution) for checksums that include the payload.
+     // Checksum engine only supports 16-bit ones' complement checksums, also known
+     // as csum16 or internet checksum.
+     // -----------------------------------------------------------------------------
+     /// Add data to checksum.
+     /// @param data : List of fields to be added to checksum calculation. The
+     /// data must be byte aligned.
+     {"Checksum.add",
+      {"data"},
+      [](const ExternMethodImpls::ExternInfo &externInfo) {
+          // We do not calculate the checksum for now.
+          return nullptr;
+      }},
+     /// Subtract data from existing checksum.
+     /// @param data : List of fields to be subtracted from the checksum. The
+     /// data must be byte aligned.
+     {"Checksum.subtract",
+      {"data"},
+      [](const ExternMethodImpls::ExternInfo &externInfo) {
+          P4C_UNIMPLEMENTED("Unimplemented extern method: %1%.%2%",
+                            externInfo.externObjectRef.toString(), externInfo.methodName);
+          return nullptr;
+      }},
+     /// Subtract data from existing checksum.
```

Comment:
Ah sorry, the description is wrong, will fix that. 

And just to make sure, we don't need the equality because the checksum value itself is symbolic, so using just a boolean symbolic value is equivalent but simpler, is that correct?

Reply:
Yes, since we have no control over the computation anyway. (For now)